### ISSUE TITLE
How do I set/return the postcss result to a variable?

### ISSUE THREAD

**Issue by CodeBradley** [csstools/postcss-preset-env#88]
I have string of CSS that's rendered from Sass's renderSync function and I would like to run it through postcss-preset-env, but I can't get the CSS to return as string that I can store in a variable which I then want to inject into a pug template so that I can insert the CSS in a <style></style> tag; embedded CSS. Is this not possible when using postcss-preset-env? Here is some sample code:

postcss-preset-env@^6.0.7

```
const sass = require('node-sass');
const postcssPresetEnv = require('postcss-preset-env');
const postcss = require('postcss');

let sassString = `
    $bgcolor: #000000;
    body { background-color: $bgcolor; }
`;

let css = sass.renderSync({data:sassString}).css.toString();

let process = postcss([postcssPresetEnv()]).process(css).css;

console.log(process);
```

I've also tried using ```postcssPresetEnv.process(css)```

**Comment by jonathantneal** [comment]
Have you considered using something like [PostCSS Sass](https://www.npmjs.com/package/@csstools/postcss-sass) in combination with `insertAfter`? https://github.com/csstools/postcss-preset-env#insertbefore--insertafter

```js
require('postcss-preset-env')({
  insertAfter: {
    'nesting-rules': require('postcss-sass')()
  }
})
```

**Comment by CodeBradley** [comment]
Thanks for responding @jonathantneal. That's actually a pretty cool repo that I didn't know about so thanks for the share! This would not make the code synchronous though, correct? 

I'm doing something a bit different than what this repo is probably intended for as I'm not creating a web app per se, so let me explain a bit more. I'm working with a third party website that has a document section (somewhat similar to a WordPress page) which allows me to upload single page templates that have a particular format I have to follow. To accomplish this, I created a repo that generates the files in this format. A file after being generated would consist of embedded CSS, JS and then the HTML. [Here's a S.O. question I asked when getting started that explains it a bit better.](https://stackoverflow.com/questions/52176542/can-webpack-build-multiple-html-files-from-scss-and-pug)

So basically, I made an index.js file that does this and I need to pass a local variable to my pug template in order to insert the CSS that's needed, but I can't because this is  asynchronous and I can't set the resulting CSS in a variable.

**Comment by jonathantneal** [comment]
> How do I set/return the postcss result to a variable?

To answer the title of the post:

```js
import postcssPresetEnv from 'postcss-preset-env';

const { css } = await postcssPresetEnv.process(
	':root { --color: blue; color: var(--color); }',
	{ from: 'path/for/mapping/css' }
);

console.log(css); // <- PostCSS CSS result as a `css` variable
```
